Adding Unit test cases for wrappedNodeFetch and adding github actions

Hermione2408 opened the pull request:
For code coverage added unit test for wrappedNodeFetch function and integrated git hub actions.

test #354
Closes: https://github.com/keploy/keploy/issues/354

re-Tick committed:
fix: updates the commit linter versio

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 committed:
feat(github actions and wrappednodefetched.test.js): integrated github action and wrote unit test

For code coverage added unit test for wrappedNodeFetch function and integrated git hub actions.

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

developer-diganta commented:
Hi @re-Tick! Can you please review the PR?

re-Tick commented:
@Hermione2408 please resolve DCO and commitzen

Hermione2408 committed:
feat(ci.yml): revert version and removing console

Revert version and removing console from unit test file

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 commented:
Can you have a look now @re-Tick

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,108 @@
+    };
+    const response = await wrappedFetch(url, options);
+    expect(mockFetch).toHaveBeenCalledWith(url, options);
+    expect(response).toBeInstanceOf(Response);
```
Add a expect check for updated context after fetch call. to check that whether output of fetch call is added to deps/mocks or not.

Hermione2408 replied:
Added expect check @re-Tick

re-Tick replied:
Please add a check to compare the recorded output with response. This will ensure the functionality of wrappedNodeFetch.

Hermione2408 replied:
Added @re-Tick

re-Tick replied:
Awesome. LGTM. Resolving the requested changes

Hermione2408 committed:
feat(wrappednodefetch.test.ts): added expect test for mocks and deps

Added expect test for mocks and deps after fetching in record and test mode

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 committed:
feat(wrappednodefetch.test.ts): added a check to compare the recorded output with response

Added a check to compare the recorded output with response to ensure wrappedNodeFetch functionality

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,116 @@
+        {
+          Version: 'V1_BETA2',
+          Name: 'testId',
+          Kind: 'HTTP',
```
This should be "Http" because we are comparing with [this](https://github.com/keploy/typescript-sdk/blob/10ff4ebc89a3e99fecb4040cf37c6ba274556d50/src/keploy.ts#L27) constant [here](https://github.com/keploy/typescript-sdk/blob/10ff4ebc89a3e99fecb4040cf37c6ba274556d50/integrations/octokit/require.ts#L168).

Hermione2408 replied:
Got it! Thanks for the help.. have made the changes.

Hermione2408 committed:
feat(wrappednodefetch.test.ts): importing HTTP from src/keploy.ts

importing HTTP from src/keploy.ts

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 commented:
Hey @re-Tick , Can you please review the changes now

re-Tick commented:
a test is failing. I will try these changes locally and request changes if any.

Hermione2408 commented:
Sure I'll try to resolve too where they are failing

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,118 @@
+
+describe('wrappedNodeFetch', () => {
+  it('should call fetch function with correct arguments in record mode', async () => {
+    const mockFetch = jest.fn().mockResolvedValueOnce(new Response());
```
Since, outputs are recorded when end event is triggered for response of fetch. We dont need to mock fetch here. We can use the actual fetch of node-fetch

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,118 @@
+      deps: [],     
+    };
+    createExecutionContext(ctx)
+    const wrappedFetch = (wrappedNodeFetch(mockFetch) as any).bind({ fetch: mockFetch });
```
And pass actual fetch here. To capture the outputs.

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,118 @@
+    };
+    createExecutionContext(ctx)
+    const wrappedFetch = (wrappedNodeFetch(mockFetch) as any).bind({ fetch: mockFetch });
+    const url = 'http://example.com';
```
We can use keploy's health API "https://api.keploy.io/healthz" here as url

re-Tick commented on test/wrappedNodeFetch.test.ts:
```diff
@@ -0,0 +1,118 @@
+    const deps=updatedctx.deps.length;
+    const responseBody = await response.text();
+    const recordedOutput = updatedctx.mocks[0].Spec.Res.Body;
+    expect(mockFetch).toHaveBeenCalledWith(url, options);
```
We also dont need to do this since we use actual fetch of `node-fetch`

Hermione2408 committed:
test(wrappednodefetch, octokit/require.ts): adding fetch method and createExecutionContext

Resolving the failed test cases by using fetch from node-fetch and createExecutionContext

test#354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 commented:
@re-Tick I have made the changes and it is passing all the test cases now.

Hermione2408 committed:
test(wrappednodefetch): updating checks

updating test cases

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

Hermione2408 committed:
Merge branch 'main' into test#354

Hermione2408 committed:
feat(require.ts): removing createExecutionContext

removing createExecutionContext from octokit

test #354

Signed-off-by: Hermione Dadheech <hermionedadheech@gmail.com>

re-Tick commented:
LGTM. Please add this PR link in GSOC task list.
